hypertrace-core-graphql-span-schema/src/main/java/org/hypertrace/core/graphql/span/dao/SpanLogEventResponseConverter.java

```diff
@@ -0,0 +1,76 @@
+package org.hypertrace.core.graphql.span.dao;
+
+import io.reactivex.rxjava3.core.Observable;
+import io.reactivex.rxjava3.core.Single;
+import java.util.Collection;
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.inject.Inject;
+import lombok.experimental.Accessors;
+import org.hypertrace.core.graphql.attributes.AttributeStore;
+import org.hypertrace.core.graphql.atttributes.scopes.HypertraceCoreAttributeScopeString;
+import org.hypertrace.core.graphql.common.request.AttributeRequest;
+import org.hypertrace.core.graphql.common.utils.BiConverter;
+import org.hypertrace.core.graphql.context.GraphQlRequestContext;
+import org.hypertrace.core.graphql.log.event.schema.LogEvent;
+import org.hypertrace.gateway.service.v1.common.Value;
+import org.hypertrace.gateway.service.v1.log.events.LogEventsResponse;
+import org.hypertrace.gateway.service.v1.span.SpansResponse;
+
+class SpanLogEventResponseConverter {
+
+  private final BiConverter<Collection<AttributeRequest>, Map<String, Value>, Map<String, Object>>
+      attributeMapConverter;
+  private final AttributeStore attributeStore;
+
+  @Inject
+  SpanLogEventResponseConverter(
+      BiConverter<Collection<AttributeRequest>, Map<String, Value>, Map<String, Object>>
+          attributeMapConverter,
+      AttributeStore attributeStore) {
+    this.attributeMapConverter = attributeMapConverter;
+    this.attributeStore = attributeStore;
+  }
+
+  Single<SpanLogEventsResponse> buildResponse(
+      GraphQlRequestContext graphQlRequestContext,
+      Collection<AttributeRequest> attributeRequests,
+      SpansResponse spansResponse,
+      LogEventsResponse logEventsResponse) {
+    String key =
+        attributeStore
+            .getForeignIdAttribute(
+                graphQlRequestContext,
+                HypertraceCoreAttributeScopeString.LOG_EVENT,
+                HypertraceCoreAttributeScopeString.SPAN)
+            .blockingGet()
+            .key();
+    return Observable.fromIterable(logEventsResponse.getLogEventsList())
+        .concatMapSingle(
+            logEventsResponseVar -> this.convert(attributeRequests, logEventsResponseVar))
+        .collect(Collectors.groupingBy(logEvent -> (String) logEvent.attribute(key)))
+        .map(
+            spanIdVsLogEventsMap -> new SpanLogEventsResponse(spansResponse, spanIdVsLogEventsMap));
+  }
+
+  private Single<LogEvent> convert(
+      Collection<AttributeRequest> request,
+      org.hypertrace.gateway.service.v1.log.events.LogEvent logEvent) {
+    return this.attributeMapConverter
+        .convert(request, logEvent.getAttributesMap())
+        .map(v -> new ConvertedLogEvent(v));
```

Comment:
nit:
```suggestion
        .map(ConvertedLogEvent::new);
```
(or rename the var)